mypy/join.py

```diff
@@ -454,6 +468,27 @@ def object_from_instance(instance: Instance) -> Instance:
     return res
 
 
+def object_or_any_from_type(typ: ProperType) -> ProperType:
+    # Similar to object_from_instance() but tries hard for all types.
+    # TODO: find a better way to get object, or make this more reliable.
+    if isinstance(typ, Instance):
+        return object_from_instance(typ)
+    elif isinstance(typ, (CallableType, TypedDictType)):
+        return object_from_instance(typ.fallback)
+    elif isinstance(typ, TupleType):
+        return object_from_instance(typ.partial_fallback)
+    elif isinstance(typ, TypeType):
+        return object_or_any_from_type(typ.item)
+    elif isinstance(typ, TypeVarType) and isinstance(typ.upper_bound, ProperType):
+        return object_or_any_from_type(typ.upper_bound)
+    elif isinstance(typ, UnionType):
+        for it in typ.items:
+            o = object_or_any_from_type(it)
+            if isinstance(o, Instance):
+                return object_from_instance(o)
```

Comment:
This seems to pick one union item arbitrarily. It seems non-deterministic, since union item order shouldn't affect semantics. 

Random idea: Maybe a join of results of all items instead?